apps/core/mutations.py

```diff
@@ -239,6 +251,58 @@ def mutate(table, root, info, id, data: Any):
         return TableJoinMutation(result=joined_table, errors=None, ok=True)
 
 
+class JoinPreviewMutation(graphene.Mutation):
+    class Arguments:
+        data = TableJoinInputType(required=True)
+        id = graphene.ID(required=True)
+
+    errors = graphene.List(graphene.NonNull(CustomErrorType))
+    ok = graphene.Boolean()
+    result = graphene.Field(RowsColumnsType)
+
+    @staticmethod
+    @lift_mutate_with_instance(Table)
+    def mutate(source_table, root, info, id, data: Any):
+        target_table_id = data["target_table"]
+        join_type = data["join_type"]
+        clauses = data["clauses"]
+        target_table = Table.objects.get(id=target_table_id)
+
+        is_join_with_equi_clause = (
+            len(clauses) == 1
+            and clauses[0]["operation"] == JOIN_CLAUSE_OPERATIONS.EQUAL
+        )
+        if is_join_with_equi_clause:
+            new_cols, new_rows, _ = perform_hash_join_(
+                clause=clauses[0],
+                source_cols=source_table.preview_data["columns"],
+                target_cols=target_table.preview_data["columns"],
+                source_rows=source_table.preview_data["rows"],
+                target_rows=target_table.preview_data["rows"],
+                source_stats=[],
+                target_stats=[],
+                join_type=join_type,
+                conflicting_col_suffix=str(target_table.id),
+            )
+        else:
+            return DiveMutationMixin(
+                errors=[
+                    dict(
+                        field="nonFieldErrors",
+                        messages=(
+                            "This clause is not supported at the moment."
+                            " Please use single inner join with equality clause."
```

Comment:
```suggestion
                            "Please use single inner join with equality clause."
```

Reply:
Grammar rules: there is space after a period.